### PR TITLE
Add `flat` command

### LINKED ISSUE #161: `flat` command
Ranger has a command `flat` which allows a heavily nested folder to be browsed as a flattened list.
I think this is a nice feature to add to Joshuto; ideally it would be possible to bulk-rename files in this format too.

## Example

```
A/
    B.txt
    C/
        D.txt
        E.txt
```

Is currently displayed as:

```
A
```

With flat mode enabled, it would be displayed like this:

```
A/B.txt
A/C/D.txt
A/C/E.txt
```

### PR DESCRIPTION
**Pull request by krivahtoo**
closes #161

### CODE REVIEW

File: src/commands/flat.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            if options.show_hidden() {
+                true
+            } else {
+                !_is_hidden(e)
+            }
+        })
+        .filter(|e| {
+            !e.as_ref()
+                .ok()
+                .unwrap()
```
Comment (kamiyaa) [inline src/commands/flat.rs:37]: This crashes
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/commands/flat.rs:37:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
I think the cause for this is when you don't have enough permissions for specific folders. ie. `/root`.

**Review by kamiyaa — CHANGES_REQUESTED** [review]
Looks good overall, just a minor bug.

Thanks!

File: src/commands/flat.rs (hunk 1 above)
Comment (krivahtoo) [reply src/commands/flat.rs]: fixed

**Comment by kamiyaa** [comment]
Thanks!

**Pull request merged**